### PR TITLE
Editor Utility Tools Update

### LINKED ISSUE #824: Tilegenerator creates icons which then result in an "Failed to build appx from Solution.  Error 1"
I used the Tilegenerator to make all the icons.   When I went to build via Holotoolkit, the AppX build failed -- red text flashed by and I got an error 1.

Running MSBuild from the command line brought up:

`"C:\dev\AugmntrHoloTest\WindowsStoreApp\AugmntrHoloTest.sln" (default target) (1) ->
"C:\dev\AugmntrHoloTest\WindowsStoreApp\AugmntrHoloTest\AugmntrHoloTest.csproj" (default target) (2) ->
(_ValidateAppxPackage target) ->
  Package.appxmanifest(19,54): error APPX3207: App manifest references the wide 310x150 logo image 'Assets\Wide310x150Logo.scale-400.png' which is larger than the maximum image file size. It must be no larger than 204,800 bytes.`

The outputted icon is 236KB.

`[C:\dev\AugmntrHoloTest\WindowsStoreApp\AugmntrHoloTest\AugmntrHoloTest.csproj]
  Package.appxmanifest(20,27): error APPX1619: App manifest references the splash screen image 'Assets\SplashScreen.scale-400.png' which does not have valid dimensions. It must be 2480x1200 pixels. [C:\dev\AugmntrHoloTest\WindowsStoreApp\
AugmntrHoloTest\AugmntrHoloTest.csproj]`

The outputted icon is 2400x1200.

### LINKED ISSUE #856: Icon Generator does not support transparency
Icon Generator does not support transparency.

### LINKED ISSUE #893: Build module can't find msbuild for VS 2017
The underlying process call to cmd.exe is returning the error "'vswhere' is not recognized as an internal or external command". Verified that vswhere.exe is in the installation directory. Call works fine when I open up a command window and paste it in. Just not working from System.Diagnostics.Process .

### LINKED ISSUE #906: BuildDeployTools Launch Application returns 500 error code.
Build Window -> Launch Application returns 500 error code when attempting to launch application.

```
WebException: The remote server returned an error: (500) Internal Server Error.
```

### LINKED ISSUE #919: BuildWindow sometimes opens out of screen with 2 monitors running on different DPI
After strugling a bit with my BuildWindow not appearing whenever I clicked it from the HoloToolkit menu I found out that it had gone out of my screen.

To get it back I had to change the DPI of my main monitor to 100% after which it would appear somewhere on my right screen again.

To reproduce the issue:

1. Open Unity + BuildWindow on screen running on 100% dpi
2. Drag Unity to screen running 200% dpi
3. Start dragging buildwindow from screen with 100% dpi to screen with 200% dpi. It now disappears into infinity. (When you now release your mouse it's gone).

I'm not sure if this an issue that can be solved in the MRT, but if we can do something like whenever you open the window that it will appear in a visible position, that would be great.

### PR DESCRIPTION
**Pull request by StephenHodgson**
- Fixes #906 Launching App from Build window returns 500 error code.
- Fixes #856 Icon Generator does not support transparency.
- Fixes #824 Does not assign images if over maximum byte size limit.
- Fixes #893, #903 Finding `msbuild.exe` path for vs 2017. Determines whether machine is x86 or x64 to get right path to `vswhere.exe`.  If more than one vs 2017 sku is found, pull enterprise first, then pro, then community.
- Fixes #919 Build window now opens next to the game and scene windows.
- Added a centralized `EditorPrefsUtility` and updated all `EditorPrefs` with new usage of a unique application key so projects don't share settings.
- Moved `CrossPlatform` folder into `Utilities/Scripts` folder.
- Removed duplicated text serialization option.
- Fixes unused modified path string in `OpenLocalFileEditor` and `OpenLocalFolderEditor`.
- Fixes issue with unreadable dark text in config windows with Editor Pro Skin enabled.
- Fixed `HACK` code for updating project and quality settings in `Apply Project Settings` window.
- Updated Capabilities window to disable `InternetClient` if `InternetClientServer` or `PrivateInternetClientServer` is enabled.
- Adds options for Sharing Service configuration in HoloLens Project Settings Config Window.
capture
- Adds options for Sharing Service capabilities in HoloLens Capabilities Config Window.
capture

### CODE REVIEW

**Comment by Zod-** [comment]
I've recently found these 2 potential bugs in `OpenLocalFileEditor` and `OpenLocalFolderEditor` where the result of a path replace does not get used.
https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/fedaa19fe3294004f9e9717d3b8de030aac960d2/Assets/HoloToolkit/Utilities/Scripts/Editor/OpenLocalFileEditor.cs#L38-L41
https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/fedaa19fe3294004f9e9717d3b8de030aac960d2/Assets/HoloToolkit/Utilities/Scripts/Editor/OpenLocalFolderEditor.cs#L38-L41

**Comment by StephenHodgson** [comment]
Thanks @Zod- I'll be sure to add these to the update.

File: Assets/HoloToolkit/Utilities/Scripts/Editor/CapabilitySettingsWindow.cs [hunk 1]
```diff
@@ -30,6 +31,8 @@ protected override void ApplySettings()
             ApplySetting(PlayerSettings.WSACapability.SpatialPerception);
             ApplySetting(PlayerSettings.WSACapability.WebCam);
             ApplySetting(PlayerSettings.WSACapability.InternetClient);
+            ApplySetting(PlayerSettings.WSACapability.InternetClientServer);
+            ApplySetting(PlayerSettings.WSACapability.PrivateNetworkClientServer);
```
Comment (keveleigh) [inline Assets/HoloToolkit/Utilities/Scripts/Editor/CapabilitySettingsWindow.cs]: What is this capability used for in the Toolkit that isn't covered by InternetClientServer?

Comment (StephenHodgson) [reply Assets/HoloToolkit/Utilities/Scripts/Editor/CapabilitySettingsWindow.cs]: This goes back to our prev conversation in the update PR haha.

I'll apply those learnings.

Comment (keveleigh) [reply Assets/HoloToolkit/Utilities/Scripts/Editor/CapabilitySettingsWindow.cs]: Sounds good!

Comment (StephenHodgson) [reply Assets/HoloToolkit/Utilities/Scripts/Editor/CapabilitySettingsWindow.cs]: The [documentation for win 10 capabilities](https://docs.microsoft.com/en-us/uwp/schemas/appxpackage/uapmanifestschema/element-uap-capability) hasn't been updated recently. Keeps [sending me back to win 8](https://msdn.microsoft.com/library/windows/apps/hh464936).

Comment (keveleigh) [reply Assets/HoloToolkit/Utilities/Scripts/Editor/CapabilitySettingsWindow.cs]: Weird, yeah, that's definitely a stale link. [Here's the Win10 documentation.](https://docs.microsoft.com/en-us/windows/uwp/devices-sensors/enable-device-capabilities)

Comment (keveleigh) [reply Assets/HoloToolkit/Utilities/Scripts/Editor/CapabilitySettingsWindow.cs]: Weird, yeah, that's definitely a stale link. [Here's the Win10 documentation.](https://docs.microsoft.com/en-us/windows/uwp/packaging/app-capability-declarations)

**Comment by StephenHodgson** [comment]
Can I get a second review on this?  I'd like to get it in before publishing a new release.

**Comment by StephenHodgson** [comment]
@Zod- or @jamesashley1 would either of you like to submit a review?

**Pull request merged**